### PR TITLE
TINY-9385: Identify if an element is scrollable

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: 
https://ephocks.atlassian.net/browse/TINY-9385

Description of Changes:
- added scroller detection to cover all edge cases such as `overflow-x`, `overflow-y`

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [ ] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: modules/tinymce/src/themes/silver/test/ts/headless/modes/ScrollingContextTest.ts [hunk 1]
```diff
@@ -34,6 +34,26 @@ describe('headless.modes.ScrollingContextTest', () => {
       const div = SugarElement.fromHtml('<div style="overflow: scroll;">A</div>');
       assert.isTrue(ScrollingContext.isScroller(div), 'Should be a scroller');
     });
+
+    it('TINY-9226: overflow-x: scroll - a scroller', () => {
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/test/ts/headless/modes/ScrollingContextTest.ts:38]: Might be worth to add test cases for `overflow:hidden`?

Comment (tiny-ben-tran) [reply modules/tinymce/src/themes/silver/test/ts/headless/modes/ScrollingContextTest.ts:38]: According to https://developer.mozilla.org/en-US/docs/Web/CSS/overflow, `overflow:clip` is a possible case

File: modules/tinymce/src/themes/silver/test/ts/headless/modes/ScrollingContextTest.ts [hunk 2]
```diff
@@ -34,6 +44,26 @@ describe('headless.modes.ScrollingContextTest', () => {
       const div = SugarElement.fromHtml('<div style="overflow: scroll;">A</div>');
       assert.isTrue(ScrollingContext.isScroller(div), 'Should be a scroller');
     });
+
+    it('TINY-9385: overflow-x: scroll - a scroller', () => {
```
Comment (tiny-ben-tran) [inline modules/tinymce/src/themes/silver/test/ts/headless/modes/ScrollingContextTest.ts:48]: `overflow-x` and `overflow-y` have `hidden` and `clip` values https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-x. I suggest that more tests are needed to cover those values, and the cases where either `overflow-x` or `overflow-y` is `hidden` and `clip`.

File: modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts [hunk 3]
```diff
@@ -14,12 +14,23 @@ export interface ScrollingContext {
...
+
+    // If overflow-x and overflow-y have different values, ignore 'overflow'
+    if (overflowX !== overflowY) {
+      return (Strings.trim(overflowX).length > 0 && !Arr.contains(nonScrollingOverflows, overflowX)) ||
+        (Strings.trim(overflowY).length > 0 && !Arr.contains(nonScrollingOverflows, overflowY));
+    }
+
+    return (Strings.trim(overflow).length > 0 && !Arr.contains(nonScrollingOverflows, overflow)) ||
+      (Strings.trim(overflowX).length > 0 && !Arr.contains(nonScrollingOverflows, overflowX)) ||
+      (Strings.trim(overflowY).length > 0 && !Arr.contains(nonScrollingOverflows, overflowY));
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts:33]: Since we are using `Css.get` which gets computed values we shouldn't need to also check `'overflow'` as when `'overflow: <value>'` is set in the DOM, the 'overflow-x' and 'overflow-y' properties are also set.

```
const isScrollingOverflowValue = (value: string): boolean =>
  Strings.trim(value).length > 0 && !Arr.contains(nonScrollingOverflows, value);
```

```suggestion
    const overflowX = Css.get(elem, 'overflow-x');
    const overflowY = Css.get(elem, 'overflow-y');
    return isScrollingOverflowValue(overflowX) || isScrollingOverflowValue(overflowY);
```

File: modules/tinymce/CHANGELOG.md [hunk 4]
```diff
@@ -34,6 +34,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Exposed `dataTransfer` property of drag and drop events for elements with a `contenteditable="false"` attribute. #TINY-9601
 - Screen readers now announce instructions for resizing the editor using arrow keys, when the resize handle is focused. #TINY-9793
 - Dialog `tabpanel` tab labels are now allowed to word wrap for better readability with long labels #TINY-9947
+- Improved isScroller function to include elements with overflow-x or overflow-y set to scroll or auto as scrollable. #TINY-9385
```
Comment (ltrouton) [inline modules/tinymce/CHANGELOG.md:37]: The changelog entry needs to be more relevant to the customer. It looks like at a higher level, this improves the `ui_mode: 'split'` option
```suggestion
- Improved detection of scrollable containers when the `ui_mode: 'split'` option is set. #TINY-9385
```

File: modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts [hunk 5]
```diff
@@ -14,12 +14,23 @@ export interface ScrollingContext {
 // TINY-9385: Investigate exactly what makes an element possibly
 // scrollable. This approach does not seem exhaustive. What about:
 // overflow-x, overflow-y etc.
```
Comment (ltrouton) [inline modules/tinymce/src/themes/silver/main/ts/modes/ScrollingContext.ts:16]:
```suggestion
// See https://developer.mozilla.org/en-US/docs/Glossary/Scroll_container for what makes an element scrollable
```

**Review by ltrouton — APPROVED** [review]

**Review by tiny-ben-tran — APPROVED** [review]

**Review by TheSpyder — APPROVED** [review]
